### PR TITLE
Add safe math operations to AUM

### PR DESCRIPTION
**Pull request by nullpointer0x00**
## Summary

Adds safe math to the AUM fee calculations so bad inputs return an error instead of failing in an uncontrolled way.

## Changes

- `CalculateAUMFee` now recovers from arithmetic overflow and returns a clear error with the offending inputs.
- `PerformVaultFeeTransfer` uses `SafeAdd` when combining the new fee with the outstanding balance, returning an error on overflow.

## Tests

- Added cases for overflow and edge inputs in `interest_test.go` and `reconcile_test.go`.

## Summary by CodeRabbit

* **Bug Fixes**
  * AUM calculations now safely handle arithmetic overflow by returning errors instead of panicking.
  * Vault fee transfer operations now use safe arithmetic to prevent overflow failures.

* **Tests**
  * Expanded AUM fee calculation tests to cover edge cases including negative values and overflow scenarios.
  * Added tests for vault fee transfer overflow error handling.

### CODE REVIEW

File: keeper/reconcile.go [hunk 1]
```diff
@@ -259,7 +259,11 @@ func (k Keeper) PerformVaultFeeTransfer(ctx sdk.Context, vault *types.VaultAccou
 		return fmt.Errorf("failed to calculate accrued AUM fee payment: %w", err)
 	}
 
-	totalOutstanding := vault.OutstandingAumFee.Add(newFeePayment)
+	totalOutstandingAmount, err := vault.OutstandingAumFee.Amount.SafeAdd(newFeePayment.Amount)
+	if err != nil {
+		return fmt.Errorf("failed to add new fee payment %s to outstanding AUM fee %s: %w", newFeePayment, vault.OutstandingAumFee, err)
+	}
+	totalOutstanding := sdk.NewCoin(vault.PaymentDenom, totalOutstandingAmount)
```
[the remark this answers is not in the document]
Comment (nullpointer0x00) [reply keeper/reconcile.go:266]: The denom consistency here is already guaranteed on both operands, so I don't think this change introduces a silent-mismatch risk:

- `newFeePayment` is built by `CalculateAccruedAUMFeePayment`, which constructs its coin with `sdk.NewCoin(vault.PaymentDenom, …)` in both return branches (`reconcile.go:409` and `:415`) — its denom is always `PaymentDenom` by construction.
- `vault.OutstandingAumFee` is initialized with `PaymentDenom` (`types/vault.go:122`) and the `VaultAccount.Validate()` invariant enforces `OutstandingAumFee.Denom == PaymentDenom` (`types/vault.go:287-288`).

So rebuilding the result from `vault.PaymentDenom` is correct, and the drift this guards against can only exist if validation is bypassed.

The dropped panic is also intentional. This runs in the fee-collection/reconcile path, where a panic on denom mismatch is a chain-halt vector,  moving to amount-level `SafeAdd` is the goal of this PR series. If we wanted an explicit guard, it should return an error rather than panic, but given the `Validate()` invariant it would be effectively dead code. Leaving as-is.

File: interest/interest_test.go [hunk 2]
```diff
@@ -188,6 +189,39 @@ func TestCalculateAUMFee(t *testing.T) {
 	}
 }
 
+// TestCalculateAUMFeeOverflow proves the recover guard turns a LegacyDec overflow into an
+// error instead of panicking.
+func TestCalculateAUMFeeOverflow(t *testing.T) {
```
Comment (Taztingo) [inline interest/interest_test.go]: Do you think it would help to move this into the other test, but just as new cases where we check the error output?

File: keeper/reconcile.go [hunk 3]
```diff
@@ -259,7 +259,11 @@ func (k Keeper) PerformVaultFeeTransfer(ctx sdk.Context, vault *types.VaultAccou
 		return fmt.Errorf("failed to calculate accrued AUM fee payment: %w", err)
 	}
 
-	totalOutstanding := vault.OutstandingAumFee.Add(newFeePayment)
+	totalOutstandingAmount, err := vault.OutstandingAumFee.Amount.SafeAdd(newFeePayment.Amount)
```
Comment (Taztingo) [inline keeper/reconcile.go:262]: I noticed `totalOutstanding.Sub(toCollect)` on line 297 doesn't use `SafeSub` like this `SafeAdd`. Unsure if it's needed or wanted for consistency, and maybe the previous checks might already guarantee this is safe.

Comment (nullpointer0x00) [reply keeper/reconcile.go:262]: Even though, this would probably never panic.  I will add it just to be consistent.  I think having a pattern of doing math this way in critical code areas will help future development.

**Review by Taztingo — APPROVED** [review]
